File: JavaIntroNeuralNetworkEdition2/src/com/heatonresearch/httprecipes/spider/workload/WorkloadException.java
package com.heatonresearch.httprecipes.spider.workload;

/**
 * The Heaton Research Spider 
 * Copyright 2007 by Heaton Research, Inc.
 * 
 * HTTP Programming Recipes for Java ISBN: 0-9773206-6-9
 * http://www.heatonresearch.com/articles/series/16/
 * 
 * WorkloadException: This exception is thrown when the
 * workload manager encounters an error.
 * 
 * This class is released under the:
 * GNU Lesser General Public License (LGPL)
 * http://www.gnu.org/copyleft/lesser.html
 * 
 * @author Jeff Heaton
 * @version 1.1
 */
public class WorkloadException extends Exception {

	/**
	 * Serial id for this class.
	 */
  private static final long serialVersionUID = 1L;

  /**
   * Construct a WorkloadException with the specified
   * message.
   * 
   * @param message
   *          The exception message.
   */
  public WorkloadException(String message) {
    super(message);
  }

  /**
   * Called to wrap another exception as a
   * WorkloadException.
   * 
   * @param t
   *          An exception to be wrapped.
   */
  public WorkloadException(Throwable t) {
    super(t);
  }
}

File: CSharpHTTPRecipesEdition1/HTTPRecipes/HeatonResearchSpider/Spider/Workload/WorkloadException.cs
// The Heaton Research Spider for .Net 
// Copyright 2007 by Heaton Research, Inc.
// 
// From the book:
// 
// HTTP Recipes for C# Bots, ISBN: 0-9773206-7-7
// http://www.heatonresearch.com/articles/series/20/
// 
// This class is released under the:
// GNU Lesser General Public License (LGPL)
// http://www.gnu.org/copyleft/lesser.html
//
using System;
using System.Collections.Generic;
using System.Text;

namespace HeatonResearch.Spider.Workload
{
    /// <summary>
    /// WorkloadException: This exception is thrown when the
    /// workload manager encounters an error.
    /// </summary>
    public class WorkloadException:Exception
    {
        /// <summary>
        /// Throw a workload exception.
        /// </summary>
        /// <param name="msg">The exception message.</param>
        public WorkloadException(String msg)
            : base(msg)
        {
        }

        /// <summary>
        /// Pass on an exception as a WorkloadException.
        /// </summary>
        /// <param name="e">The other exception.</param>
        public WorkloadException(Exception e)
            : base("Exception while processing workload", e)
        {
        }
    }
}
